### ISSUE TITLE
Can't destroy using modules on 0.7rc01

### ISSUE THREAD

**Issue by dbcoliveira** [hashicorp/terraform#7047]
Hi there,

I am currently using terraform 0.7rc01 and I can't destroy something that have been applied using modules (docker provider) just seconds before.

A snippet of the module source goes like:

```
variable docker_host {}
variable container_name {}
variable servers {}
variable arguments {}

provider "docker" {
    host = "tcp://${var.docker_host}:2375/"
}
.
.
.
```

A snippet of the module goes as:
```
variable "ipaddress" {
        default = "10.64.78.75,10.64.78.76,10.64.78.77"
}

module "etcd01" {
  source = "module/etcd/"
  docker_host = "${element(split(",",var.ipaddress),0)}"
  container_name = "${var.cname}"
  servers = "${var.servers}"
  arguments = ""
}
.
.
.
```

So applying with:

`terraform apply -target=module.etcd01 -target=module.etcd02 -target=module.etcd03 -var-file=secret.tfvars
`

```
module.etcd02.docker_container.etcd-viewer: Creation complete
module.etcd03.docker_container.etcd-viewer: Creation complete

Apply complete! Resources: 9 added, 0 changed, 0 destroyed.
```

Everything goes well and confirmed that it really applies.
On the destroy part it fails big:

`terraform destroy -target=module.etcd01 -target=module.etcd02 -target=module.etcd03 -var-file=secret.tfvars `
```
Do you really want to destroy?
  Terraform will delete the following infrastructure:
  	module.etcd01
  	module.etcd02
  	module.etcd03
  There is no undo. Only 'yes' will be accepted to confirm

  Enter a value: yes

.........
Error applying plan:

3 error(s) occurred:

* 1:9: unknown variable accessed: var.docker_host in:

tcp://${var.docker_host}:2375/
* 1:9: unknown variable accessed: var.docker_host in:

tcp://${var.docker_host}:2375/
* 1:9: unknown variable accessed: var.docker_host in:

tcp://${var.docker_host}:2375/
```

Note that I know the variable list declaration implementation was introduced on 0.7 so I've actually try to perform the same code without the ipaddress list but using the actual ips. the result is the same.

Any ideas ?

Thank you for the feedback

**Comment by phinze** [comment]
Hi @dbcoliveira - thanks for the report, and sorry for the trouble!

Here is a full repro config that I have confirmed works in 0.6.16 and does not work on 0.7.0-rc1 or master:

```tf
# main.tf
variable "in" {
  default = "1,2,3,4"
}

module "foo" {
  source = "./foo"
  in = "${element(split(",", var.in), 0)}"
}

output "out" {
  value = "${module.foo.out}"
}
```

```
# foo/foo.tf
variable "in" { }

provider "aws" {
  region = "us-east-${var.in}"
}

resource "aws_security_group" "bar" {
  tags {
    In = "${var.in}"
  }
}

output "out" {
  value = "${aws_security_group.bar.id}"
}
```

Steps:

```
terraform apply
terraform destroy -force
```

Expected:

Happy destroy

Got:

```
Error applying plan:

1 error(s) occurred:

* 1:11: unknown variable accessed: var.in in:

us-east-${var.in}
```

**Comment by phinze** [comment]
Just confirmed the list is not required - simplifying the example above to reflect that.

**Comment by phinze** [comment]
cc @jen20 FYI - planning on digging into this later but in case you get to it first. 👍

**Comment by phinze** [comment]
Fixed in #7131 👍
